tests/unit/problemTest.php

```diff
@@ -0,0 +1,144 @@
+        $testCaseDTOs = array(new TestCaseFactoryDTO('rightComand', $ExecutionRuleDTOs));
+        Problem::Create('rightTitle', 'rightBoby', 0, 1, $compileRuleDTOs, $testCaseDTOs);
+    }
+
+    public function test_memoryConstraint_overflow()
+    {
+        $this->expectException(InvalidMemoryConstraintException::class);
+        $langueage = Language::C;
+        $compileRuleDTOs = array(new CompileRuleFactoryDTO($langueage, 'rightComand', 'rightComand'));
+        $ExecutionRuleDTOs = array(new ExecutionRuleFactoryDTO($langueage, 'right', 'right', 'right', 'right'));
+        $testCaseDTOs = array(new TestCaseFactoryDTO('rightComand', $ExecutionRuleDTOs));
+        Problem::Create('rightTitle', 'rightBoby', 1, 2 * 1024 * 1024+1, $compileRuleDTOs, $testCaseDTOs);
+    }
+
+    public function test_memoryConstraint_underflow()
+    {
+        $this->expectException(InvalidMemoryConstraintException::class);
+        $langueage = Language::C;
+        $compileRuleDTOs = array(new CompileRuleFactoryDTO($langueage, 'rightComand', 'rightComand'));
+        $ExecutionRuleDTOs = array(new ExecutionRuleFactoryDTO($langueage, 'right', 'right', 'right', 'right'));
+        $testCaseDTOs = array(new TestCaseFactoryDTO('rightComand', $ExecutionRuleDTOs));
+        Problem::Create('rightTitle', 'rightBoby', 1, 0, $compileRuleDTOs, $testCaseDTOs);
+    }
+
+
+
+    public function test_is_disableFlag_faluse_whenProblemCreate()
+    {
+        $langueageC = Language::C;
+        $compileRuleDTOs = array(new CompileRuleFactoryDTO($langueageC, 'rightComand', 'rightComand'));
+        $ExecutionRuleDTOs = array(new ExecutionRuleFactoryDTO($langueageC, 'right', 'right', 'right', 'right'));
+        $testCaseDTOs = array(new TestCaseFactoryDTO('rightComand', $ExecutionRuleDTOs));
+        $problem = Problem::Create('rightTitle', 'rightBoby', 1, 200, $compileRuleDTOs, $testCaseDTOs);
+        $testCases = $problem->getTestCases();
+        foreach ($testCases as $testCase) {
+            $this->assertFalse($testCase->getIsDisabled());
+        }
+    }
+
+    public function test_one_language_have_one_compileRule()
+    {
+        $this->expectException(InvalidDTOException::class);
+
+        $langueageC = Language::C;
+        // １つの言語につき、2つのcompileRuleがあるため例外が投げられる
+        $compileRuleDTOs = array(
+            new CompileRuleFactoryDTO($langueageC, 'rightComand1', 'rightComand1'),
+            new CompileRuleFactoryDTO($langueageC, 'rightComand2', 'rightComand2')
+        );
+
+        $ExecutionRuleDTOs = array(new ExecutionRuleFactoryDTO($langueageC, 'right', 'right', 'right', 'right'));
+        $testCaseDTOs = array(new TestCaseFactoryDTO('rightComand', $ExecutionRuleDTOs));
+        $problem = Problem::Create('rightTitle', 'rightBoby', 1, 200, $compileRuleDTOs, $testCaseDTOs);
+    }
+    public function test_one_language_have_one_executionRule()
```

Comment:
```suggestion
    public function test_multiple_executionrules_with_the_same_language_at_instantiation()
```

同じテストをインスタンス作成時のみだけでなく、後からProblemに追加する場合にもテストする必要がある